test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
-@test B*A ≈ B*Array(A)
-
-# complex matrix-vector multiplication and left-division
-if Base.USE_GPL_LIBS
-for i = 1:5
-    a = speye(5) + 0.1*sprandn(5, 5, 0.2)
-    b = randn(5,3) + im*randn(5,3)
-    c = randn(5) + im*randn(5)
-    d = randn(5) + im*randn(5)
-    α = rand(Complex128)
-    β = rand(Complex128)
-    @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
-    @test (maximum(abs.(A_mul_B!(similar(b), a, b) - Array(a)*b)) < 100*eps()) # for compatibility with present matmul API. Should go away eventually.
-    @test (maximum(abs.(A_mul_B!(similar(c), a, c) - Array(a)*c)) < 100*eps()) # for compatibility with present matmul API. Should go away eventually.
-    @test (maximum(abs.(At_mul_B!(similar(b), a, b) - Array(a).'*b)) < 100*eps()) # for compatibility with present matmul API. Should go away eventually.
-    @test (maximum(abs.(At_mul_B!(similar(c), a, c) - Array(a).'*c)) < 100*eps()) # for compatibility with present matmul API. Should go away eventually.
-    @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
-    @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
-    @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
-    @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
-    @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
-    @test (maximum(abs.((a'*c + d) - (Array(a)'*c + d))) < 1000*eps())
-    @test (maximum(abs.((α*a.'*c + β*d) - (α*Array(a).'*c + β*d))) < 1000*eps())
-    @test (maximum(abs.((a.'*c + d) - (Array(a).'*c + d))) < 1000*eps())
-    c = randn(6) + im*randn(6)
-    @test_throws DimensionMismatch α*a.'*c + β*c
-    @test_throws DimensionMismatch α*a.'*ones(5) + β*c
-
-    a = speye(5) + 0.1*sprandn(5, 5, 0.2) + 0.1*im*sprandn(5, 5, 0.2)
-    b = randn(5,3)
-    @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
-    @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
-    @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
-    @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
-    @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
-    @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
-
-    a = speye(5) + tril(0.1*sprandn(5, 5, 0.2))
-    b = randn(5,3) + im*randn(5,3)
-    @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
-    @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
-    @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
-    @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
-    @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
-    @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
-
-    a = speye(5) + tril(0.1*sprandn(5, 5, 0.2) + 0.1*im*sprandn(5, 5, 0.2))
-    b = randn(5,3)
-    @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
-    @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
-    @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
-    @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
-    @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
-    @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
```

Comment:
Either preserve the function name ("spdiagm") or expand the description ("construction of sparse diagonal matrices")?